tests/datastore_redis/test_asyncio.py

```diff
@@ -129,24 +129,32 @@ async def _test_pipeline(client):
 @background_task()
 def test_async_pubsub(client, loop):
     messages_received = []
+    message_received = asyncio.Event()
 
     async def reader(pubsub):
         while True:
             message = await pubsub.get_message(ignore_subscribe_messages=True)
             if message:
+                message_received.set()
                 messages_received.append(message["data"].decode())
                 if message["data"].decode() == "NOPE":
                     break
 
+    async def _publish(client, channel, message):
+        """Publish a message and wait for the reader to receive it."""
+        await client.publish(channel, message)
+        await message_received.wait()
```

Comment:
Is it worth adding a timeout on this call with `asyncio.wait_for()`?

Reply:
Done